src/Workspaces/Remote/Core/Services/SolutionCreator.cs

```diff
@@ -1,20 +1,29 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+#if DEBUG
+#if false // https://github.com/dotnet/roslyn/issues/33476
+#define VALIDATE_CHECKSUM
+#endif
+#endif
+
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Linq;
-using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Diagnostics;
-using Microsoft.CodeAnalysis.Internal.Log;
 using Microsoft.CodeAnalysis.PooledObjects;
-using Microsoft.CodeAnalysis.Remote.DebugUtil;
-using Microsoft.CodeAnalysis.Remote.Shared;
 using Microsoft.CodeAnalysis.Serialization;
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
+
+#if VALIDATE_CHECKSUM
```

Comment:
Why all the fancy conditional compilation? I assume we'd be fixing this ASAP?

Reply:
> Why all the fancy conditional compilation?

It excludes the implementation helper from release builds.

> I assume we'd be fixing this ASAP?

Definitely hoping we do. It seems like a serious issue.